apis/dataset_metadata/dataset_subject.py

```diff
@@ -26,23 +27,53 @@ class DatasetSubjectResource(Resource):
     @api.response(400, "Validation Error")
     # @api.param("id", "The dataset identifier")
     @api.marshal_with(dataset_subject)
-    def get(self, study_id: int, dataset_id: int):
+    def get(self, study_id: int, dataset_id: int):  # pylint: disable= unused-argument
         dataset_ = model.Dataset.query.get(dataset_id)
         dataset_subject_ = dataset_.dataset_subject
         return [d.to_dict() for d in dataset_subject_]
 
+    @api.doc("update subject")
+    @api.response(200, "Success")
+    @api.response(400, "Validation Error")
     def post(self, study_id: int, dataset_id: int):
+        study_obj = model.Study.query.get(study_id)
+        if not is_granted("dataset_metadata", study_obj):
+            return "Access denied, you can not make any change in dataset metadata", 403
         data: Union[Any, dict] = request.json
         data_obj = model.Dataset.query.get(dataset_id)
-        dataset_subject_ = model.DatasetSubject.from_data(data_obj, data)
-        model.db.session.add(dataset_subject_)
+        list_of_elements = []
+        for i in data:
+            if "id" in i and i["id"]:
+                dataset_subject_ = model.DatasetSubject.query.get(i["id"])
+                if not dataset_subject_:
+                    return f"Study link {i['id']} Id is not found", 404
+                dataset_subject_.update(i)
+                list_of_elements.append(dataset_subject_.to_dict())
+            elif "id" not in i or not i["id"]:
+                dataset_subject_ = model.DatasetSubject.from_data(data_obj, i)
+                model.db.session.add(dataset_subject_)
+                list_of_elements.append(dataset_subject_.to_dict())
         model.db.session.commit()
-        return dataset_subject_.to_dict()
-
-    @api.route("/study/<study_id>/dataset/<dataset_id>/metadata/subject/<subject_id>")
-    class DatasetSubjectUpdate(Resource):
-        def put(self, study_id: int, dataset_id: int, subject_id: int):
-            dataset_subject_ = model.DatasetSubject.query.get(subject_id)
-            dataset_subject_.update(request.json)
-            model.db.session.commit()
-            return dataset_subject_.to_dict()
+        return list_of_elements
+
+
+@api.route("/study/<study_id>/dataset/<dataset_id>/metadata/" "subject/<subject_id>")
```

Comment:
incorrect uri?

Reply:
Not typo, all of them happened due to pylint and format. it complained first about length of the line, so i made them 2 lines instead of 1, but later looks like it made again format and glued 2 lines together. I have to fix them now again